### PR TITLE
Issue #SC-000 : Dedup changes to dup check on specific pdata  pids

### PR DESCRIPTION
**Pull request by RevathiKotla**
@SanthoshVasabhaktula Please review

### CODE REVIEW

File: ansible/roles/samza-jobs/defaults/main.yml [hunk 1]
```diff
@@ -55,6 +55,8 @@ telemetry_extractor_messages_fetch_threshold: 3000
 consumer_fetch_max_bytes: 1572864
 producer_max_request_size_bytes: 1572864
 
+dedup_include_producer_pids: sunbird-portal
```
Comment (SanthoshVasabhaktula) [inline ansible/roles/samza-jobs/defaults/main.yml]: We should use pdata.id rather than pid. Also pdata.id is env specific.

Comment (RevathiKotla) [reply ansible/roles/samza-jobs/defaults/main.yml]: changed code to use pdata.id

File: ansible/roles/samza-jobs/defaults/main.yml [hunk 2]
```diff
@@ -55,6 +55,8 @@ telemetry_extractor_messages_fetch_threshold: 3000
 consumer_fetch_max_bytes: 1572864
 producer_max_request_size_bytes: 1572864
 
+dedup_include_producer_ids: "{{producer_id_env_prefix}}.diksha.portal,{{producer_id_env_prefix}}.sunbird.desktop"
```
Comment (SanthoshVasabhaktula) [inline ansible/roles/samza-jobs/defaults/main.yml]: Move this to env specific yml file.

File: data-pipeline/de-duplication/src/main/java/org/ekstep/ep/samza/service/DeDuplicationService.java [hunk 3]
```diff
@@ -36,8 +36,8 @@ public void process(DeDuplicationSource source, DeDuplicationSink sink) throws E
 				sink.toSuccessTopic(event);
 				return;
 			}
-			if (!((null != event.producerId() && config.producerId().contains(event.producerId()))
-					|| config.serverEventEid().contains(event.eid()))) {
+
+			if (isDupCheckRequired(event)) {
```
Comment (SanthoshVasabhaktula) [inline data-pipeline/de-duplication/src/main/java/org/ekstep/ep/samza/service/DeDuplicationService.java]: Default behaviour should not break. If the inclusion/exclusion filter is not provided, all the events should be deduped.

**Pull request merged**